### PR TITLE
If any dependencies of the current build fail, and all dependants of …

### PR DESCRIPTION
**Pull request by ahmedlawi92**
…that build as having been rejected.

This will prevent them from being built, as they will fail if their dependencies have already failed.

### CODE REVIEW

**Comment by ahmedlawi92** [comment]
@matejonnet Would you mind having a look at this when you get a chance.

**Comment by ahmedlawi92** [comment]
test this please

**Comment by slaskawi** [comment]
test this please

File: build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildTask.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void setStatus(BuildStatus status) {
         log.debug("Updating build task {} status to {}", this.getId(), buildStatusChanged);
         buildSetTask.taskStatusUpdated(buildStatusChanged);
         buildStatusChangedEvent.fire(buildStatusChanged);
-        if (status.equals(BuildStatus.DONE)) {
```
Comment (slaskawi) [inline build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildTask.java]: Putting it onther way around will save you from NPE `BuildStatus.DONE.equals(status)`

File: spi/src/main/java/org/jboss/pnc/spi/builddriver/BuildDriverStatus.java [hunk 2]
```diff
@@ -25,7 +25,21 @@
  * Created by <a href="mailto:matejonnet@gmail.com">Matej Lazar</a> on 2014-11-23.
  */
 public enum BuildDriverStatus {
-    SUCCESS, FAILED, UNSTABLE, BUILDING, ABORTED, CANCELLED, UNKNOWN;
+    SUCCESS, FAILED(true), UNSTABLE, BUILDING, ABORTED(true), CANCELLED(true), UNKNOWN;
```
Comment (matejonnet) [inline spi/src/main/java/org/jboss/pnc/spi/builddriver/BuildDriverStatus.java]: Aborted and canceled doesn't have errors. Maybe we should rename field to successfullyCompleted.

Comment (ahmedlawi92) [inline spi/src/main/java/org/jboss/pnc/spi/builddriver/BuildDriverStatus.java]: An UNSTABLE build would have completed successfully right?

File: spi/src/main/java/org/jboss/pnc/spi/builddriver/BuildDriverStatus.java [hunk 3]
```diff
@@ -25,7 +25,21 @@
  * Created by <a href="mailto:matejonnet@gmail.com">Matej Lazar</a> on 2014-11-23.
  */
 public enum BuildDriverStatus {
-    SUCCESS, FAILED, UNSTABLE, BUILDING, ABORTED, CANCELLED, UNKNOWN;
+    SUCCESS(true), FAILED, UNSTABLE(true), BUILDING, ABORTED, CANCELLED, UNKNOWN;
```
Comment (matejonnet) [inline spi/src/main/java/org/jboss/pnc/spi/builddriver/BuildDriverStatus.java]: I would treat UNSTABLE as unsuccessful. It's true that build results can be consumed by others, but if you requested to run tests they should pass.

**Comment by slaskawi** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by slaskawi** [comment]
retest this please

**Comment by slaskawi** [comment]
The code looks good but it seems it needs a rebase. I'll merge it right after the rebase.

**Comment by ahmedlawi92** [comment]
retest this please

**Comment by slaskawi** [comment]
retest this please

**Comment by slaskawi** [comment]
retest this please

**Pull request merged**